tools/auto_update_v8.ts

```diff
@@ -0,0 +1,92 @@
+const V8_TRACKING_BRANCH = "9.1-lkgr";
+const AUTOROLL_BRANCH = "autoroll";
+
+function extractVersion() {
+  const MAJOR_PREFIX = "#define V8_MAJOR_VERSION ";
+  const MINOR_PREFIX = "#define V8_MINOR_VERSION ";
+  const BUILD_PREFIX = "#define V8_BUILD_NUMBER ";
+  const PATCH_PREFIX = "#define V8_PATCH_LEVEL ";
+
+  const versionDotH = Deno.readTextFileSync("./v8/include/v8-version.h");
+  const lines = versionDotH.split("\n");
+  const major = parseInt(lines.find((s) => s.startsWith(MAJOR_PREFIX))!
+    .substring(MAJOR_PREFIX.length));
+  const minor = parseInt(lines.find((s) => s.startsWith(MINOR_PREFIX))!
+    .substring(MINOR_PREFIX.length));
+  const build = parseInt(lines.find((s) => s.startsWith(BUILD_PREFIX))!
+    .substring(BUILD_PREFIX.length));
+  const patch = parseInt(lines.find((s) => s.startsWith(PATCH_PREFIX))!
+    .substring(PATCH_PREFIX.length));
+
+  return `${major}.${minor}.${build}.${patch}`;
+}
+
+await run(["git", "branch", "-f", AUTOROLL_BRANCH, "origin/main"]);
+await run(["git", "checkout", AUTOROLL_BRANCH]);
+await run(["git", "reset", "origin/main", "--hard"]);
```

Comment:
Just:
```suggestion
await run(["git", "checkout", "origin/main"]);
```
Would work because you're pushing from HEAD to a named branch. Whatever you prefer though.